stdlib/public/runtime/MutexWin32.cpp

```diff
@@ -16,6 +16,13 @@
 //===----------------------------------------------------------------------===//
 
 #if defined(_WIN32)
+
+// Notes: swift::fatalError is not shared between libswiftCore and libswift_Concurrency
+// and libswift_Concurrency uses swift_Concurrency_fatalError instead.
+#ifndef SWIFT_FATAL_ERROR
```

Comment:
It's unfortunate that this `#ifndef`/`#define` dance is scattered in several places, but that's really a symptom of the odd way in which we've included these `.cpp` files.

Reply:
Yes, it's not clear and very messy, so I hope this workaround should be a temporary solution until we will come up with a cleaner way to share impls.